src/Transaction.purs

```diff
@@ -0,0 +1,44 @@
+module Transaction
+  ( ModifyTxError(..)
+  , attachDatum
+  ) where
+
+import Prelude
+
+import Control.Monad.Except.Trans (ExceptT(ExceptT), runExceptT)
+import Data.Either (Either(Right), note)
+import Data.Generic.Rep (class Generic)
+import Data.Show.Generic (genericShow)
+import Effect (Effect)
+import Helpers (liftEither)
+import Serialization.WitnessSet as Serialization.WitnessSet
+import Deserialization.WitnessSet as Deserialization.WitnessSet
+import Serialization.PlutusData as Serialization.PlutusData
+import Types.PlutusData (Datum(Datum))
+import Types.Transaction (Transaction(Transaction))
+
+data ModifyTxError
+  = ConvertWitsError
```

Comment:
Wits -> Witns? It ConvertWitsError somehow doesn't ring "Witness(es)" in my head. If it suits the others I'll be fine.

Reply:
I think I've seen "wits" as an abbreviation for witnesses elsewhere. I'll just put `ConvertWitnessesError` though, it's clearer